### PR TITLE
UI: Display current dB with the volume meter

### PR DESCRIPTION
**Pull request by cg2121**
This adds the ability for the volume meter to display the current dB. If the dB is greater than 0 dB then CLIP will be displayed.

### CODE REVIEW

File: UI/volume-control.cpp [hunk 1]
```diff
@@ -63,6 +64,13 @@ void VolControl::VolumeLevel(float mag, float peak, float peakHold, bool muted)
 	}
 
 	volMeter->setLevels(mag, peak, peakHold);
+
+	float db = log_def_to_db(peakHold);
```
Comment (fryshorts) [inline UI/volume-control.cpp]: Please note that the volume meter might (in the future) use another algorithm to map deflection to dB. That being said it should be safe to assume that 0.0f dB yields 1.0f deflection and vice versa.
You could therefore just check for `peakHold == 1.0f` without converting it to dB.

File: libobs/obs-audio-controls.h [hunk 2]
```diff
@@ -259,6 +259,8 @@ EXPORT void obs_volmeter_add_callback(obs_volmeter_t *volmeter,
 EXPORT void obs_volmeter_remove_callback(obs_volmeter_t *volmeter,
 		obs_volmeter_updated_t callback, void *param);
 
+EXPORT float log_def_to_db(const float def);
```
Comment (fryshorts) [inline libobs/obs-audio-controls.h]: This function is supposed to be internal and not necessarily the one used be the fader/volmeter to convert between dB and deflection. The function that is used is depends on `obs_fader_type`.

**Comment by cg2121** [comment]
I have updated it to now allow for different fader types.

**Comment by ONSBalder** [comment]
Can one of the admins verify this patch?

**Comment by wallrik** [comment]
Is it possible to change the patch to use UI styles instead of hard coded red? errorLabel parhaps?

**Comment by cowai** [comment]
@jp9000  Can this please be merged?

**Comment by RytoEX** [comment]
Has @wallrik's [comment above](https://github.com/jp9000/obs-studio/pull/738#issuecomment-277851207) been addressed?  Or is there some reason not to do what is suggested there?

**Comment by jp9000** [comment]
The PR isn't currently functioning properly.  Besides, I had some issues with it from a UX standpoint.  Please do not pressure me to merge code.

**Comment by jp9000** [comment]
I had some issues with multiple sets of numbers being displayed.  I feel like it sort of made it confusing to users about what number means what, plus it increased the required paint updates to the UI which I don't really prefer (I'd prefer to keep UI painting to a minimum for performance reasons).

However, the clip-detection is very useful, and that's something that I agree the users definitely need to have.

So, what I did instead was I made it so the meter will turn in to a red gradient if it detects audio clipping.  This way it doesn't require any extra UI changes or extra UI updating, and quite clearly gets the message across to the user that it's clipping/peaking.

**Pull request closed without merging**